src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -7,30 +7,35 @@ import React, {
   useState,
 } from 'react';
 import PropTypes from 'prop-types';
+import useFetch from '@bloodyaugust/use-fetch';
 import Select from '../select/select.jsx';
 import ListOption from '../list-option/list-option.jsx';
+import mockConstants from '../../mocks/mock-constants.js';
+
+const { API_ROOT } = mockConstants;
 
 const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleChoice(props, ref) {
+  const [choices, setChoices] = useState([]);
+  const [listOptionRefs, setListOptionRefs] = useState([]);
   const [value, setValue] = useState(props.value);
   const backupRef = useRef();
   const selfRef = ref || backupRef;
-
-  const listOptionRefs = props.choices.map(() => createRef());
+  const { execute } = useFetch();
 
   function selectOnChangeHandler(event) {
     if (event.target.value) {
-      setValue(props.choices.find(choice => choice.id === event.target.value.id));
+      setValue(choices.find(choice => choice.id === event.target.value.id));
     } else {
       setValue(undefined);
     }
   }
 
   const listOptions = () => {
-    const choices = {};
-    props.choices.forEach((item, index) => { choices[item.id] = { ...item, index }; });
+    const mungedChoices = {};
+    choices.forEach((item, index) => { mungedChoices[item.id] = { ...item, index }; });
 
-    return props.choices
-      .map(item => choices[item.id])
+    return choices
+      .map(item => mungedChoices[item.id])
```

Comment:
What is going on with this `forEach` -> `map`?

We use the `forEach` to create a map: `id` -> `{ item, index: i }`
Then use the map to set it is as `{ item, index: i }`.

Can we just not do the `forEach`?

```suggestion
      .map((choice, index) => ({ ...choice, index }))
```